### PR TITLE
Create Admin anon proxy vendor account pages

### PR DESCRIPTION
**Pull request by DeeTheDev**
- Create `AnonProxy::VendorAccount` API routes for list, details
- Create anon proxy vendor account list, and details pages
- Fix table row key ids bugs in different pages
- Test

### Vendor account navigation tab icon
Screenshot 2024-01-17 at 12 44 46 PM

### Vendor account list page
Screenshot 2024-01-17 at 12 44 59 PM

### Vendor account details page
Screenshot 2024-01-17 at 12 45 17 PM

### Member details (vendor account section)
Screenshot 2024-01-17 at 12 44 21 PM

### CODE REVIEW

**Comment by rgalanakis** [comment]
@DeeTheDev VendorAccountConfigurations need to be a separate resource.

File: adminapp/src/components/EligibilityConstraints.jsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import api from "../api";
```
Comment (DeeTheDev) [inline adminapp/src/components/EligibilityConstraints.jsx:1]: Nothing changed in this component except for the prop names. It was taken from the OfferingDetailPage and made into its own component.

File: adminapp/src/pages/MemberDetailPage.jsx [hunk 2]
```diff
@@ -472,7 +474,7 @@ function MessageDeliveries({ messageDeliveries }) {
       title="Message Deliveries"
       headers={["Id", "Created", "Sent", "Template", "To"]}
       rows={messageDeliveries}
-      keyRowAttr="id"
```
Comment (rgalanakis) [inline adminapp/src/pages/MemberDetailPage.jsx:475]: This should not need to change- if it does, there's a but with `keyRowAttr`. Please check it out.

File: adminapp/src/pages/MemberDetailPage.jsx [hunk 3]
```diff
@@ -484,6 +486,34 @@ function MessageDeliveries({ messageDeliveries }) {
...
+      title="Vendor Accounts"
+      headers={[
+        "Id",
+        "Created",
+        "Vendor",
+        "Latest Access Code Magic Link",
+        "Latest Access Code",
+      ]}
+      rows={vendorAccounts}
+      getKey={(row) => row.id}
```
Comment (rgalanakis) [inline adminapp/src/pages/MemberDetailPage.jsx:501]: Same here, use `keyRowAttr`

File: adminapp/src/pages/MemberDetailPage.jsx (hunk 2 above)
Comment (rgalanakis) [reply adminapp/src/pages/MemberDetailPage.jsx:475]: Or get rid of `keyRowAttr`, but don't keep it around if it doesn't work.

File: adminapp/src/pages/VendorAccountDetailPage.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import React from "react";
+
+export default function VendorAccountDetailPage() {
+  return (
+    <ResourceDetail
+      apiGet={api.getVendorAccount}
+      title={(model) => `Vendor Account ${model.id}`}
+      properties={(model) => [
+        { label: "ID", value: model.id },
+        { label: "Created At", value: dayjs(model.createdAt) },
```
Comment (rgalanakis) [inline adminapp/src/pages/VendorAccountDetailPage.jsx:18]: I think these all need `.format`, are we doing this elsewhere? It'd use the default dayjs formatting which may not be what we want (let's be consistent with time formatting).

**Pull request merged**